### PR TITLE
Fixed case when pre-release is in suffix

### PR DESCRIPTION
**Pull request by rahul2393**
Tests added to cover scenarios when version contains pre-release in minor or patch version

### CODE REVIEW

File: pkg/scanner/utils/utils.go [hunk 1]
```diff
@@ -64,6 +65,29 @@ func FormatPatchVersion(version string) string {
 		if _, err := strconv.Atoi(part[2]); err == nil {
 			version = strings.Join(part[:3], ".") + "-" + strings.Join(part[3:], ".")
 		}
+	} else {
+		r := regexp.MustCompile(`(?P<Number>^[0-9]+)(?P<PreRelease>[a-z]*.*)`)
```
Comment (knqyf263) [inline pkg/scanner/utils/utils.go:69]: We should define this regexp outside of the function for optimization.

File: pkg/scanner/utils/utils.go [hunk 2]
```diff
@@ -64,6 +65,29 @@ func FormatPatchVersion(version string) string {
...
+			if res == nil {
+				continue
+			}
+			names := r.SubexpNames()
+			regexGroup := map[string]string{}
+			for j := range names {
+				if names[j] != "" {
+					regexGroup[names[j]] = res[j]
+				}
+			}
```
Comment (knqyf263) [inline pkg/scanner/utils/utils.go:81]: We already handle `res == nil`, so I think we don't need to use `SubexpNames`.

```suggestion
			number := res[1]
			preRelease := res[2]
```

Comment (rahul2393) [reply pkg/scanner/utils/utils.go]: @knqyf263  thanks, changes done

**Comment by knqyf263** [comment]
Could you try `$ trivy image knqyf263/vuln-image:1.2.3` and make sure there is no error?

**Comment by rahul2393** [comment]
Screenshot 2020-07-21 at 4 50 00 PM
1 error because the constraint is empty, otherwise no errors
@knqyf263

**Comment by knqyf263** [comment]
Yes, an error makes a user wonder if the result is incorrect. We have to handle the error as well.

**Comment by rahul2393** [comment]
@knqyf263 updated error to debug

**Comment by knqyf263** [comment]
Can we skip the case when the constraint is empty and suppress the error?

**Comment by rahul2393** [comment]
Do you mean
```
if err != nil {
			if v != "" {
				log.Logger.Error("NewConstraint", "error", err)
			}
			continue
		}
```

**Comment by rahul2393** [comment]
@knqyf263 please check now

**Pull request merged**